amp/WEB-INF/src/org/digijava/module/aim/action/CompareActivityVersions.java

```diff
@@ -95,643 +60,35 @@ public ActionForward compare(ActionMapping mapping, ActionForm form, HttpServlet
-                                }
-                            }
-                        }
-                        if (coincidence == 0) {
-                            // Check if the object was added in the
-                            // previous iteration.
-                            if (!auxList.contains(auxVersionable2)) {
-                                CompareOutput auxOutput = new CompareOutput(auxAnnotation.fieldTitle(), new String[] {
-                                        "",
-                                        ActivityVersionUtil.generateFormattedOutput(request, auxVersionable2
-                                                .getOutput()) }, fields[i], new Object[] { null, auxObject2 }, false,
-                                        false);
-                                vForm.getOutputCollection().add(auxOutput);
-                            }
-                        }
-                    }
-                } else {
-                    output.setStringOutput(new String[] { auxResult1 != null ? auxResult1.toString() : "",
-                            auxResult2 != null ? auxResult2.toString() : "" });
-                    output.setDescriptionOutput(auxAnnotation.fieldTitle());
-                    vForm.getOutputCollection().add(output);
-                }
-            }
+            return new ActionForward(mapping.findForward("reload").getPath() + "?ampActivityId=" + activityId, true);
         }
 
-        Map<String, List<CompareOutput>> outputGroupped = groupOutputCollection (vForm.getOutputCollection());
-        modifyFundingOutputs (outputGroupped);
-        vForm.setOutputCollectionGrouped(outputGroupped);
-    
-        return mapping.findForward("forward");
-    }
-
-    private void modifyFundingOutputs (Map<String, List<CompareOutput>> outputGroupped) {
-
-
-    }
-
-    private Map<String, List<CompareOutput>> groupOutputCollection (List<CompareOutput> outputCollection) {
-        Map<String, List<CompareOutput>> retVal = new HashMap<String, List<CompareOutput>>();
-        int idx = 0;
-        for (CompareOutput obj: outputCollection) {
-            if (!obj.getBlockSingleChangeOutput()) {
-                obj.setIndex(idx);
-                idx ++;
-            } else {
-                obj.setIndex(-1); // skip from merge process
-            }
-            if (!retVal.containsKey(obj.getDescriptionOutput())) {
-                retVal.put(obj.getDescriptionOutput(), new ArrayList<CompareOutput>());
-            }
-            retVal.get(obj.getDescriptionOutput()).add(obj);
-
-        }
-
-        return retVal;
-    }
-
-    public ActionForward enableMerge(ActionMapping mapping, ActionForm form, HttpServletRequest request,
-            HttpServletResponse response) throws Exception {
+        vForm.setOutputCollection(new ArrayList<CompareOutput>());
+        vForm.setOutputCollectionGrouped(ActivityVersionUtil.compareWithPrevious(vForm.getActivityOneId()));
```

Comment:
this shouldn't have been put here since we will use this method from the audit logger, and activity comparison from activity preview should remain the same, comparing what we have as parameter. as a test is fine (if you want to test the code) but we should not commit this since its changing the functionality that its used from preview

Reply:
That's to test whether the code works or not... we can also remove it when we migrate the code to the audit logger... I want to know your feedback on the code that compares with its previous version.

Reply:
I don't still get any significant or insignificant wrong snippet on my code. It just worked dramatically without any further step to dig... hope you can debug and check or I'm not understanding you...

Reply:
Sorry I didn't see the change, somehow GitHub didn't show. Ill provide my feedback now. However @yosephtesfaye code has already been merged so you can continue. But I think its valuable for everybody to provide feedback on every task.

As a general rule we don't commit code that then will be removed and we will reject commits with such code. If we need to add it to test, we can do it locally and then commit the code without test code. Its very difficult to track down code that will be removed or changed afterwards and with this you are changing the code that is working now.